cranelift-codegen/meta/src/shared/instructions.rs

```diff
@@ -1159,6 +1159,27 @@ pub fn define(
         .can_load(true),
     );
 
+    ig.push(
+        Inst::new(
+            "fill_nop",
+            r#"
+        This is identical to :inst:`fill`, except it has no encoding, since
+        it is a no-op.
+
+        This instruction is created only during late-stage peephole
+        optimisation, after all registers and stack slots have been assigned.
+        It is used to replace :inst:`fill`s that have been identified as
+        redundant on the basis that a dominating :inst:`fill` or :inst:`spill`
```

Comment:
I might end this paragraph after "redundant" since the reasons for the redundancy are not interesting here.  As a case in point, I started using the copy_nop in my simple register allocator for reasons unrelated to its original use.